document/document_test.go

```diff
@@ -357,10 +360,13 @@ func TestNewFromStruct(t *testing.T) {
 	q := 5
 	u.Q = &q
 	u.R = new(group)
+	u.S = u.R
```

Comment:
Could you explain this change? It was originally to test the behavior with nil fields.

Reply:
The tests wasn't totally clear, I didn't know if it was wanted or not. Because some case was not handled.
I will change it.

Reply:
I agree this test is not clear, I will add a few comments in the future